### PR TITLE
zadanie 2

### PR DESCRIPTION
**Pull request by a-janczak**

### CODE REVIEW

File: ML_zadanie.ipynb [hunk 1]
```diff
@@ -0,0 +1,882 @@
...
+      "source": [
+        "from sklearn.neighbors import KNeighborsClassifier\n",
+        "\n",
+        "for nazwa, m in [\n",
+        "    (\"Dummy      \", DummyClassifier(strategy='uniform')),\n",
+        "    (\"Drzewo     \", DecisionTreeClassifier(max_depth=3)),\n",
+        "    (\"KNN (k=5)  \", KNeighborsClassifier(n_neighbors=5)),\n",
+        "]:\n",
+        "    w = cross_val_score(m, X, y, cv=5)\n",
+        "    print(f\"{nazwa} {w.mean():.3f} +/- {w.std():.3f}\")"
```
Comment (Dmytro-Lifyentsyev) [inline ML_zadanie.ipynb:784]: Też brakuje mi komentarza tutaj, ponieważ całe badanie skupia się na metryce f1-score, a tutaj porównujesz modele pod względem accuracy.

**Review by Dmytro-Lifyentsyev — COMMENTED** [review]
Wydaje mi się, ze prawie wszystkie wymagania zadania zostały spełnione, oprócz wniosków końcowych (podsumowanie całego badania).

Podoba mi się, że zwizualizowałaś badania hiperparametrów i wyciągnełaś z tego wnioski.

**Review by gorkowykamil — COMMENTED** [review]
Bardzo dobre przygotowanie danych (i oryginalne). Fajnie, że użyłaś/eś kroswalidacji oraz podajesz wnioski cząstkowe i wyjaśnienia. Kod jest dość czytelny. Problemem jest jednak nieprzeskalowanie danych przed treningiem modelu KNN. Kolumna Fare, która posiada 3-cyfrowe wartości ma przez to większe znaczenie niż np. Family members (1-cyfrowe wartości). Efekt widać po metrykach, których wartości są niższe niż przy Decision Tree. Poza tym, według mnie, warto by było zamiast lub dodatkowo oprócz hiperparametru weights badać rodzaj używanej metryki (euklidesowa, kosinusowa). Na tym bazuje ten model (odległości), a jak widać po pracy mojej i jeszcze jednej osoby z naszej grupy, wynik okazał się być różny. Brakuje mi jeszcze wniosków podsumowujących, które porównałyby wszystkie modele i spróbowały odpowiedzieć na pytanie dlaczego ten jest najlepszy, a nie inny. Poza tym, wymagania zostały spełnione, dobra praca.

**Review by JonaszKubsik — COMMENTED** [review]
Można by było dorobić kilka rzeczy:
- przeskalować dane do KNN,
- dodać wykres dla parametru weights, aby pokazać, że 'distance' jest faktycznie lepszy, na tym samym wykresie mogłyby być krzywe dla jego 2 wartości,
- wnioski końcowe.

Na plus:
- tabela z danymi, nie widziałem jeszcze u nikogo podobnych kolumn jak 'HasCabin' lub te z tytułami
- dokładne opisy co teraz robisz, co widać w tabelach itp.

**Comment by slovik02** [comment]
Wykonanie zadania ogólnie oceniam na ok. Dobrze, że porównujesz dwa warianty DummyClassifiera i tłumaczysz różnicę, imo dobry sposób na naukę ogółem. 

Idąc dalej to z minusów trochę za późno został zrobiony train/test split. Uzupełnianie danych dzieje się na całym titanic_data, co oznacza, że mediana i dominanta którą liczysz, 'widzi' też dane testowe, czyli mamy wyciek. Nie za bardzo widzę też potrzebę robienia 2x tego splita w kodzie (zwłaszcza że ten drugi nie ma random state). 

Fajnie że zauważasz w markdownie, że KNN jako model odległościowy potrzebuje standaryzacji, bo Fare ma zupełnie inną skalę niż zmienne zero-jedynkowe - szkoda że ta obserwacja nie trafiła do kodu ;p

No i na koniec krótkie podsumowanie który model wygrał i dlaczego tak mogło być
